### PR TITLE
Update docker dependencies to fix issues

### LINKED ISSUE #360: Traefik docker backend error
The traefik container is logging few errors even if the service is apparently behaving correctly (the dashboard is available and the nodes are added to the default lb pool). 

**I have the following setup:**
Centos 7.2 
Docker 1.9.1
Traefik version: v1.0.0-beta.666

**Traefik configuration:**
port = ":80"
logLevel = "DEBUG"
[web]
address = ":8888"

[docker]
endpoint = "unix:///var/run/docker.sock"
domain = "my.domain.com"
watch = true

**I have seen the following error message in the logs:**
`level=error msg="Docker connection error Error response from daemon: json: cannot unmarshal object into Go value of type []string, retrying in 58.133390694s`

What this error is about?

### PR DESCRIPTION
**Pull request by vdemeester**
Update `engine-api`, `libcompose` and related dependencies (including `libkermit`), to fix issues with docker v1.9.1 and lower 🐮.

Should fix #360

/cc @emilevauge @gdippolito 

🐸

Signed-off-by: Vincent Demeester <vincent@sbr.pm>

### CODE REVIEW

File: build.Dockerfile [hunk 1]
```diff
@@ -6,7 +6,7 @@ RUN go get github.com/Masterminds/glide \
 && go get github.com/kisielk/errcheck
 
 # Which docker version to test on
-ENV DOCKER_VERSION 1.10.1
+ENV DOCKER_VERSION 1.9.1
```
Comment (vdemeester) [inline build.Dockerfile]: This is to make sure it works but I wonder if I should keep it or not in the end :angel:

File: integration/integration_test.go [hunk 2]
```diff
@@ -47,7 +47,7 @@ func (s *BaseSuite) TearDownSuite(c *check.C) {
 }
 
 func (s *BaseSuite) createComposeProject(c *check.C, name string) {
-	projectName := fmt.Sprintf("integration-test-%s", name)
+	projectName := fmt.Sprintf("integrationtest%s", name)
```
Comment (vdemeester) [inline integration/integration_test.go]: Same here, this change should not be needed anymore (fixed it in `libkermit`)

File: build.Dockerfile (hunk 1 above)
Comment (emilevauge) [inline build.Dockerfile]: Maybe we should use multiple versions in travis tests: https://docs.travis-ci.com/user/environment-variables/#Global-Variables

Comment (vdemeester) [inline build.Dockerfile]: Yeah, `libcompose` and `libkermit` are tested against multiple daemon version for example.

**Comment by emilevauge** [comment]
Once a solution is found to this issue https://github.com/travis-ci/travis-ci/issues/929, looks OK :)

**Pull request merged**